app/controllers/concerns/authorization.rb

```diff
@@ -1,7 +1,7 @@
 module Authorization
   def authorize!
     render file: 'layouts/login' and return if must_login
-    render_not_found if !from_login_callback?  && !can_visit?
+    raise Exceptions::OrganizationPrivateException if !from_login_callback? && !can_visit? && !from_logout?
```

Comment:
I'd try to be consistent here: we are raising 404 errors using a `render_not_found` message, but raising 403 error by explicitly throwing an exception. 

I think we should always use the same strategy, whichever it is. I think that raising the exception instead of wrapping it in a method is in the end easier since it allows you to raise it from models. 

Thus, I 'd just inline that render_not_found, and raise another custom exception in the `Exception` namespace mapped to 404, but that actually means something more specific, like `Exceptions::ContentNotAccessibleException` or something like that.